integraality/tests/test_app.py

```diff
@@ -80,14 +80,18 @@ class QueriesTests(PagesProcessorTests):
 
     def setUp(self):
         super().setUp()
-        patcher = patch('pages_processor.PropertyStatistics', autospec=True)
-        self.mock_property_statistics = patcher.start()
-        self.addCleanup(patcher.stop)
+        patcher1 = patch('pages_processor.PropertyStatistics', autospec=True)
+        self.mock_property_statistics = patcher1.start()
+        self.addCleanup(patcher1.stop)
+        patcher2 = patch('pages_processor.PropertyStatistics.GROUP_MAPPING', autospec=True)
+        self.mock_group_mapping = patcher2.start()
+        self.addCleanup(patcher2.stop)
 
     def test_queries_success(self):
         self.mock_pages_processor.return_value.make_stats_object_for_page_title.return_value = self.mock_property_statistics  # noqa
         self.mock_property_statistics.get_query_for_items_for_property_positive.return_value = "X"
         self.mock_property_statistics.get_query_for_items_for_property_negative.return_value = "Z"
+        self.mock_group_mapping.return_value = 'Q2'
```

Comment:
This confused me since we don't expect mapping to contain the grouping value other than for special cases. Closer to the truth would be to have this mock raise an Error as a side_effect

Reply:
Yes, much better idea :) Done